service/ui_plugin_build_baron.go

```diff
@@ -49,6 +53,44 @@ func bbGetConfig(settings *evergreen.Settings) map[string]evergreen.BuildBaronPr
 	return projects
 }
 
+func bbGetTask(taskId string, execution string) (*task.Task, error) {
+	oldId := fmt.Sprintf("%v_%v", taskId, execution)
+	t, err := task.FindOneOld(task.ById(oldId))
+	if err != nil {
+		return t, err
+	}
+	// if the archived task was not found, we must be looking for the most recent exec
+	if t == nil {
+		t, err = task.FindOne(task.ById(taskId))
+		if err != nil {
+			return nil, err
+		}
+	}
+	return t, nil
+}
+
+func (uis *UIServer) bbGetTaskAndBFSuggestionClient(taskId string, execution string) (*task.Task, bfSuggestionClient, error) {
+	bfsc := bfSuggestionClient{}
+	t, err := bbGetTask(taskId, execution)
+	if err != nil {
+		return nil, bfsc, err
+	}
+
+	bbProj, ok := uis.buildBaronProjects[t.Project]
+	if !ok {
+		err = errors.Errorf("Build Baron project for %v not found", t.Project)
+		return nil, bfsc, err
```

Comment:
you can merge lines 81 and 82, no need to assign first. 

same with 87

Reply:
done